test/dymaptic.GeoBlazor.Core.Test.Blazor.Shared/Components/FeatureLayerTests.razor

```diff
@@ -470,4 +470,361 @@
+                                  ObjectIdField="OBJECTID"/>
+                </Map>
+            </MapView>);
+        await WaitForMapToRender();
+
+        var q = await layer!.CreateQuery();
+        q.ReturnGeometry = true;
+        q.OutFields = ["*"];
+
+        var count = await layer.QueryFeatureCount(q);
+        Assert.IsTrue(count >= 0);
+
+        var extent = await layer.QueryExtent(q);
+        Assert.IsNotNull(extent);
+
+        var topQ = await layer.CreateQuery();
+        topQ.ReturnGeometry = true;
+        topQ.OutFields = ["State", "TOTAL", "F2018", "F2019", "F2020", "Park"];
+        topQ.OrderByFields = ["TOTAL DESC"];
+        topQ.Num = 3;
+
+        var topResult = await layer.QueryFeatures(topQ);
+        Assert.IsNotNull(topResult);
+        Assert.IsTrue(topResult!.Features?.Count > 0);
+
+        var clone = await layer.Clone();
+        Assert.IsNotNull(clone);
+
+        FeatureLayer? relLayer = null;
+        AddMapRenderFragment(
+            @<MapView class="map-view" OnViewRendered="renderHandler">
+                <Map>
+                    <FeatureLayer @ref="relLayer" OutFields="@(new[] { "*" })">
+                        <PortalItem PortalItemId="234d2e3f6f554e0e84757662469c26d3" />
+                    </FeatureLayer>
+                </Map>
+                <Extent Xmax="-13620669.8431" Xmin="-13640432.281" Ymax="4556710.618000001" Ymin="4536523.6511999965">
+                    <SpatialReference Wkid="102100" />
+                </Extent>
+            </MapView>);
+        await WaitForMapToRender();
+
+        var relQuery = new RelationshipQuery
+        {
+            RelationshipId = 0,
+            ObjectIds      = new[] { new ObjectId(1) }
+        };
+
+        var relCounts = await relLayer!.QueryRelatedFeaturesCount(relQuery);
+        Assert.IsNotNull(relCounts);
+        Assert.IsTrue(relCounts!.Values.All(c => c >= 0));
+
+    var refreshTcs = new TaskCompletionSource<RefreshEvent>();
+
+    FeatureLayer? eventLayer = null;
+    AddMapRenderFragment(
+        @<MapView class="map-view" OnViewRendered="renderHandler">
+            <Map>
+                <FeatureLayer @ref="eventLayer"
+                              Source="@([])"
+                              OutFields="@(new[] { "*" })"
+                              ObjectIdField="OBJECTID"
+                              OnRefresh="@(EventCallback.Factory.Create<RefreshEvent>(this, e => refreshTcs.TrySetResult(e)))" />
+            </Map>
+        </MapView>);
+    await WaitForMapToRender();
+    await eventLayer!.Add(new Graphic(new Point(-117, 34, 0), new SimpleMarkerSymbol(color: new MapColor("black"))));
+
+    await eventLayer.Refresh();
+    var refreshEvent = await Task.WhenAny(refreshTcs.Task, Task.Delay(5000));
+    Assert.AreSame(refreshTcs.Task, refreshEvent);
+    }
+
+
+    [TestMethod]
+    public async Task FeatureLayer_All_Getters_Invoke_JS(Action renderHandler)
```

Comment:
Let's rename this something like `HostedFeatureLayer_Can_Call_Getters_On_All_Properties`